kubernetes/kubernetes.go

```diff
@@ -492,13 +545,55 @@ func (ka KubeAgentConfig) sendMetricsToCustomS3(metricSample *os.File) {
 	}
 }
 
-// generateSampleKey creates a key (location) for s3 to upload the sample to. Example of s3 location format
+func (ka KubeAgentConfig) sendMetricsToCustomBlob(metricSample *os.File) {
+	cred, err := azidentity.NewClientSecretCredential(ka.CustomAzureTenantID, ka.CustomAzureClientID,
+		ka.CustomAzureClientSecret, nil)
+	if err != nil {
+		log.Fatalf("Could not establish Azure credentials, "+
+			"ensure Azure environment variables are set correctly: %s", err)
+	}
+	azureClient, err := azblob.NewClient(ka.CustomAzureBlobURL, cred, nil)
+	if err != nil {
+		log.Fatalf("Could not establish Azure Client, "+
+			"ensure Azure environment variables are set correctly: %s", err)
+	}
+	ka.uploadBlob(azureClient, metricSample)
+}
+
+func (ka KubeAgentConfig) uploadBlob(client *azblob.Client, metricSample *os.File) {
+	file, err := os.Open(metricSample.Name())
+	if err != nil {
+		log.Fatalf("Unable to open metric sample file %v", err)
+	}
+	defer file.Close()
+	key := generateSampleKey(ka.clusterUID, azure)
+	_, err = client.UploadFile(context.Background(), ka.CustomAzureUploadBlobContainerName, key, file, nil)
+	if err != nil {
+		log.Fatalf("Failed to put Object to custom Azure blob with error: %s", err)
+	} else {
+		sn := strings.Split(metricSample.Name(), "/")
+		log.Infof("Exported metric sample %s to custom Azure blob: %s",
+			strings.TrimSuffix(sn[len(sn)-1], ".tgz"), ka.CustomAzureUploadBlobContainerName)
+	}
+	err = os.Remove(metricSample.Name())
+	if err != nil {
+		log.Warnf("Warning: Unable to cleanup after metric sample upload: %v", err)
+	}
+}
+
+// generateSampleKey creates a key (location) for s3 or azure to upload the sample to. Example of s3 location format
 // /production/data/metrics-agent/<YYYY>/<MM>/<DD>/<CLUSTER_UID>/<CLUSTER_UID>-<YYYYMMDD>-<HH>-<MM>.tgz
-func generateSampleKey(clusterUID string) string {
+// example of azure location format
+// production/data/metrics-agent/<YYYY>/<MM>/<DD>/<CLUSTER_UID>/<CLUSTER_UID>-<YYYYMMDD>-<HH>-<MM>.tgz
+func generateSampleKey(clusterUID, provider string) string {
 	currentTime := time.Now()
 	year, month, day := currentTime.Date()
 	hour := currentTime.Hour()
 	minute := currentTime.Minute()
+	if provider == "azure" {
```

Comment:
If the only difference is the slash it'd be cleaner to just prepend that in the s3 case rather than duplicate the format. Either via a direct string concat or a `pathPrefix` var that is `""` for azure